pkg/pillar/types/global.go

```diff
@@ -25,6 +25,82 @@ const (
 // Agents subscribe to this info to get at least the log levels
 // A value of zero means we should use the default
 // All times are in seconds.
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key			string
+	ItemType	ConfigItemType
+
+	intMin		int32
+	intMax		int32
+	intDefault	int32
+
+	stringDefault 	string
+	boolDefault 	bool
+	triStateDefault TriState
+}
+
+type ConfigItemSpecMap struct {
+	itemMap map[string]ConfigItemSpec
+}
+
+func (configMap ConfigItemSpecMap) AddIntItem(key string, defaultInt int32, min int32, max int32) {
+	log.Infof("Adding int item %s", key)
+	if defaultInt < min || defaultInt > max {
+		log.Infof("Adding int item %s failed", key)
+		return
+	}
+	configItem := ConfigItemSpec{
+		ItemType:   CONFIGITEM_TYPE_INT,
+		Key:        key,
+		intDefault: defaultInt,
+		intMin:     min,
+		intMax:     max,
+	}
+	configMap.itemMap[key] = configItem
+	log.Infof("Added int item %s", key)
+}
+
+func (configMap ConfigItemSpecMap) AddBoolItem(key string, defaultBool bool) {
+	log.Infof("Adding bool item %s", key)
```

Comment:
One log.Info is good Enough.. Hmm.. Infact.. I am wondering if we need log.Info at all?? 
we can dump the map once after all items are added? THis code is not expected to change much